cache/compressed_secondary_cache.h

```diff
@@ -86,18 +86,22 @@ class CompressedSecondaryCache : public SecondaryCache {
       64,   96,   128,  160,  192,  224,   256,   320,   384,   448,  512,
       640,  768,  896,  1024, 1280, 1536,  1792,  2048,  2560,  3072, 3584,
       4096, 5120, 6144, 7168, 8192, 10240, 12288, 14336, 16384, 32768};
+
   struct CacheValueChunk {
-    CacheAllocationPtr chunk_ptr{nullptr};
-    size_t charge{0};
-    std::unique_ptr<CacheValueChunk> next{nullptr};
+    CacheValueChunk* next;
```

Comment:
Can we allocate chunks using ```AllocateBlock``` and make this ```CacheAllocationPtr next```? It can be cast to ```CacheValueChunk*``` when traversing.

Reply:
I tried this but some weird error happened as shown in https://github.com/facebook/rocksdb/pull/10452/ 
I added one TODO for future optimization.